### ISSUE TITLE
Sync Gateway crashes while processing the changes feed

### ISSUE THREAD

**Issue by elevine** [couchbase/sync_gateway#809]
While running our app's test suite against a Sync Gateway build from c3599690e5, I consistently get the following crash:

```
2015-04-28T16:19:10.391-04:00 Changes: MultiChangesFeed done   (to 0cb04886-757c-43fe-84e3-c2eb1d034d9d)
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x88 pc=0x64503e]

goroutine 105 [running]:
github.com/couchbase/sync_gateway/db.func┬╖014()
        c:/BuildAgent/work/12217e9d79624586/src/github.com/couchbase/sync_gateway/db/changes.go:399 +0x173e
created by github.com/couchbase/sync_gateway/db.(*Database).MultiChangesFeed
        c:/BuildAgent/work/12217e9d79624586/src/github.com/couchbase/sync_gateway/db/changes.go:417 +0x5d6
```

The full log is available at https://gist.github.com/elevine/be82564f5022139094a4

**Comment by adamcfraser** [comment]
Thanks for the log details.  Do you happen to know the last build that was running clean for your app's test suite, to help isolate the problem?  There isn't anything obvious in that particular commit, although I'm still investigating.

**Comment by adamcfraser** [comment]
After looking at your logs in more detail, the crash is happening when a user is deleted via the admin API while that user still has an active _changes feed running.  It's reloading the user record (in response to the delete), and SG isn't handling the nil user record properly.  

My guess is that this was an existing bug in Sync Gateway (the possibility for crash when actively replicating users are deleted), and it's just surfacing for you now because the commit for #775 is removing previously existing latency that was hiding the problem.

**Comment by elevine** [comment]
@adamcfraser that matches with my interpretation of the logs.  I can try out the tests later on an older Sync Gateway release and report back the results.

**Comment by elevine** [comment]
@adamcfraser  I just ran the test against 1.0.3 and also got the crash.  Here is the stack trace:

```
09:41:27.523534 Cache:   Deferring #27 (4 now waiting for #20...#20)
09:41:27.524534 Changes+: MultiChangesFeed waiting...   (to 775da0f8-5b7c-408d-abbb-157f442c9763)
09:41:27.524534 Cache:   Deferring #24 (5 now waiting for #20...#20)
09:41:27.524534 Changes+: Waiting for "sync_gateway"'s count to pass 3
09:41:27.525534 Changes: MultiChangesFeed done   (to 775da0f8-5b7c-408d-abbb-157f442c9763)
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x88 pc=0x4c6ec5]

goroutine 61 [running]:
runtime.panic(0x90f040, 0x11ead6f)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/runtime/panic.c:266 +0xc8
github.com/couchbaselabs/sync_gateway/db.func┬╖010()
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/changes.go:318 +0x1045
created by github.com/couchbaselabs/sync_gateway/db.(*Database).MultiChangesFeed
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/changes.go:326 +0x41d

goroutine 1 [IO wait]:
net.runtime_pollWait(0x342938, 0x72, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/runtime/netpoll.goc:116 +0x6d
net.(*pollDesc).Wait(0xc084056ef0, 0x72, 0x340fb8, 0x3e5)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_poll_runtime.go:81 +0x37
net.(*ioSrv).ExecIO(0xc084000cc0, 0xc084056de0, 0x9ce410, 0x8, 0xc084116f00, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:188 +0x223
net.(*netFD).acceptOne(0xc084056d80, 0xacacc8, 0xc084126460, 0x2, 0x2, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:539 +0x3ab
net.(*netFD).accept(0xc084056d80, 0xacacc8, 0x0, 0x0, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:566 +0x157
net.(*TCPListener).AcceptTCP(0xc0840c6268, 0x4754ae, 0x372bf8, 0x4754ae)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/tcpsock_posix.go:233 +0x4a
net.(*TCPListener).Accept(0xc0840c6268, 0x341c10, 0xc0841121b8, 0xc084130380, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/tcpsock_posix.go:243 +0x2a
net/http.(*Server).Serve(0xc08414f3c0, 0x347b70, 0xc0840c6268, 0x0, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1622 +0x94
github.com/couchbaselabs/sync_gateway/base.ListenAndServeHTTP(0x9a6a60, 0x5, 0x0, 0x0, 0x0, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/base/http_listener.go:52 +0x3f1
github.com/couchbaselabs/sync_gateway/rest.(*ServerConfig).serve(0xc084021300, 0x9a6a60, 0x5, 0x3412b8, 0xc0840
fa180)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/config.go:390 +0xaa
github.com/couchbaselabs/sync_gateway/rest.RunServer(0xc084021300)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/config.go:430 +0x6ec
github.com/couchbaselabs/sync_gateway/rest.ServerMain()
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/config.go:436 +0x2e
main.main()
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/main.go:16 +0x1d

goroutine 27 [chan receive]:
github.com/couchbaselabs/sync_gateway/db.(*DatabaseContext).watchDocChanges(0xc0840bf8c0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/assimilator.go:16 +0xa8
created by github.com/couchbaselabs/sync_gateway/db.NewDatabaseContext
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/database.go:114 +0x33c

goroutine 28 [IO wait]:
net.runtime_pollWait(0x342bd8, 0x72, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/runtime/netpoll.goc:116 +0x6d
net.(*pollDesc).Wait(0xc08410ba70, 0x72, 0x340fb8, 0x3e5)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_poll_runtime.go:81 +0x37
net.(*ioSrv).ExecIO(0xc084000cc0, 0xc08410b960, 0x9b16a0, 0x7, 0xac98b8, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:188 +0x223
net.(*netFD).Read(0xc08410b900, 0xc084109820, 0x18, 0x18, 0x0, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:451 +0x10c
net.(*conn).Read(0xc0840fbf40, 0xc084109820, 0x18, 0x18, 0x3f, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/net.go:122 +0xc8
io.ReadAtLeast(0x342ca8, 0xc0840fbf40, 0xc084109820, 0x18, 0x18, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/io/io.go:288 +0xf9
io.ReadFull(0x342ca8, 0xc0840fbf40, 0xc084109820, 0x18, 0x18, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/io/io.go:306 +0x74
github.com/dustin/gomemcached.(*MCRequest).Receive(0xc0843d9060, 0x342ca8, 0xc0840fbf40, 0xc084109820, 0x18, ..
.)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/dustin/gomemcached/mc_req.go:129 +0xe2
github.com/dustin/gomemcached/client.(*Client).runFeed(0xc0840d9690, 0xc084008c00, 0xc0840fa9c0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/dustin/gomemcached/client/tap_feed.go:270 +0x11f
created by github.com/dustin/gomemcached/client.(*Client).StartTapFeed
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/dustin/gomemcached/client/tap_feed.go:251 +0x218

goroutine 74 [chan receive]:
net/http.(*persistConn).readLoop(0xc08431a280)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/transport.go:778 +0x692

created by net/http.(*Transport).dialConn
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/transport.go:528 +0x60a


goroutine 23 [runnable]:
time.Sleep(0x9502f900)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/runtime/time.goc:31 +0x34
github.com/couchbaselabs/sync_gateway/db.func┬╖003()
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/change_cache.go:68 +0x65
created by github.com/couchbaselabs/sync_gateway/db.(*changeCache).Init
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/change_cache.go:70 +0x1c1

goroutine 25 [chan receive]:
github.com/couchbaselabs/sync_gateway/base.func┬╖002()
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/base/bucket.go:94 +0x56
created by github.com/couchbaselabs/sync_gateway/base.couchbaseBucket.StartTapFeed
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/base/bucket.go:104 +0x1c0

goroutine 26 [chan receive]:
github.com/couchbaselabs/sync_gateway/db.func┬╖006()
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/change_listener.go:49 +0xb8
created by github.com/couchbaselabs/sync_gateway/db.(*changeListener).Start
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/change_listener.go:66 +0x261

goroutine 24 [select]:
github.com/couchbaselabs/go-couchbase.(*TapFeed).run(0xc0840d4000)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/couchbaselabs/go-couchbase/tap.go:55 +0x2fc
created by github.com/couchbaselabs/go-couchbase.(*Bucket).StartTapFeed
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/couchbaselabs/go-couchbase/tap.go:39 +0x14d

goroutine 29 [select]:
github.com/couchbaselabs/go-couchbase.(*TapFeed).forwardTapEvents(0xc0840d4000, 0xc0840fa9c0, 0xc0840089c0, 0xc
084116500, 0xf)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/couchbaselabs/go-couchbase/tap.go:101 +0x24c
created by github.com/couchbaselabs/go-couchbase.(*TapFeed).connectToNodes
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/couchbaselabs/go-couchbase/tap.go:93 +0x36a

goroutine 30 [IO wait]:
net.runtime_pollWait(0x342a88, 0x72, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/runtime/netpoll.goc:116 +0x6d
net.(*pollDesc).Wait(0xc084146bf0, 0x72, 0x340fb8, 0x3e5)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_poll_runtime.go:81 +0x37
net.(*ioSrv).ExecIO(0xc084000cc0, 0xc084146ae0, 0x9ce410, 0x8, 0xc0841f7a30, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:188 +0x223
net.(*netFD).acceptOne(0xc084146a80, 0xacacc8, 0xc084126000, 0x2, 0x2, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:539 +0x3ab
net.(*netFD).accept(0xc084146a80, 0xacacc8, 0x0, 0x0, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/fd_windows.go:566 +0x157
net.(*TCPListener).AcceptTCP(0xc084179710, 0x4754ae, 0x4252d60, 0x4754ae)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/tcpsock_posix.go:233 +0x4a
net.(*TCPListener).Accept(0xc084179710, 0x341c10, 0xc084024c68, 0xc08400c880, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/tcpsock_posix.go:243 +0x2a
net/http.(*Server).Serve(0xc084195460, 0x347b70, 0xc084179710, 0x0, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1622 +0x94
github.com/couchbaselabs/sync_gateway/base.ListenAndServeHTTP(0x9cddb0, 0xe, 0x0, 0x0, 0x0, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/base/http_listener.go:52 +0x3f1
github.com/couchbaselabs/sync_gateway/rest.(*ServerConfig).serve(0xc084021300, 0x9cddb0, 0xe, 0x3412b8, 0xc0840
e9b80)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/config.go:390 +0xaa
created by github.com/couchbaselabs/sync_gateway/rest.RunServer
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/config.go:428 +0x5d1

goroutine 54 [runnable]:
github.com/couchbaselabs/sync_gateway/rest.(*handler).sendSimpleChanges(0xc08433ca20, 0xc0843a5300, 0x0, 0x16,
0x0, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/changes_api.go:176 +0x8e7
github.com/couchbaselabs/sync_gateway/rest.(*handler).handleChanges(0xc08433ca20, 0x0, 0x0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/changes_api.go:134 +0x9a4
github.com/couchbaselabs/sync_gateway/rest.(*handler).invoke(0xc08433ca20, 0xac1e70, 0x0, 0x0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/handler.go:159 +0x35f
github.com/couchbaselabs/sync_gateway/rest.func┬╖015(0x348638, 0xc08417be60, 0xc08433d5b0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/handler.go:86 +0x7f
net/http.HandlerFunc.ServeHTTP(0xc0840df800, 0x348638, 0xc08417be60, 0xc08433d5b0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1220 +0x43
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0840a1410, 0x348638, 0xc08417be60, 0xc08433d5b0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/gorilla/mux/mux.go:86 +0x1df
github.com/couchbaselabs/sync_gateway/rest.func┬╖017(0x348638, 0xc08417be60, 0xc08433d5b0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/routing.go:203 +0xcb
net/http.HandlerFunc.ServeHTTP(0xc0840fa180, 0x348638, 0xc08417be60, 0xc08433d5b0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1220 +0x43
net/http.serverHandler.ServeHTTP(0xc08414f3c0, 0x348638, 0xc08417be60, 0xc08433d5b0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1597 +0x171
net/http.(*conn).serve(0xc084130280)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1167 +0x7ba
created by net/http.(*Server).Serve
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1644 +0x28e

goroutine 63 [runnable]:
net/http.(*persistConn).roundTrip(0xc08431a280, 0xc084292d20, 0xc08431a280, 0x0, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/transport.go:879 +0x6d9

net/http.(*Transport).RoundTrip(0xc08400c300, 0xc0843a6c30, 0x1, 0x0, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/transport.go:187 +0x394

net/http.send(0xc0843a6c30, 0x341290, 0xc08400c300, 0x0, 0x0, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/client.go:168 +0x382
net/http.(*Client).send(0x11f3880, 0xc0843a6c30, 0x84, 0x7, 0x4206bc)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/client.go:100 +0xdc
net/http.(*Client).doFollowingRedirects(0x11f3880, 0xc0843a6c30, 0xacbc80, 0x0, 0x0, ...)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/client.go:294 +0x674
net/http.(*Client).Do(0x11f3880, 0xc0843a6c30, 0xc08400c900, 0x0, 0x0)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/client.go:129 +0x92
github.com/couchbaselabs/go-couchbase.(*Bucket).ViewCustom(0xc08410b300, 0x9f0210, 0xc, 0x9eddb0, 0xb, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/couchbaselabs/go-couchbase/views.go:136 +0x43e
github.com/couchbaselabs/sync_gateway/base.couchbaseBucket.ViewCustom(0xc08410b300, 0x9f0210, 0xc, 0x9eddb0, 0x
b, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/base/bucket.go:1 +0x9a
github.com/couchbaselabs/sync_gateway/db.(*DatabaseContext).ComputeRolesForUser(0xc0840bf8c0, 0x347900, 0xc0844
9ca50, 0x1c, 0x2f6c868, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/db/crud.go:812 +0x1bf
github.com/couchbaselabs/sync_gateway/auth.(*Authenticator).rebuildRoles(0xc08433b220, 0x347900, 0xc08449ca50,
0x347900, 0xc08449ca50)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/auth/auth.go:155 +0x69
github.com/couchbaselabs/sync_gateway/auth.func┬╖003(0xc08431c904, 0x105, 0x105, 0x0, 0xc0842179b0, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/auth/auth.go:114 +0x281
github.com/couchbaselabs/go-couchbase.(*Bucket).update(0xc08410b300, 0xc0843a5870, 0x2f, 0x0, 0xc08433b240, ...
)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/couchbaselabs/go-couchbase/client.go:565 +0xaf
github.com/couchbaselabs/go-couchbase.(*Bucket).Update(0xc08410b300, 0xc0843a5870, 0x2f, 0x0, 0xc08433b240, ...
)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/couchbaselabs/go-couchbase/client.go:556 +0x57
github.com/couchbaselabs/sync_gateway/base.couchbaseBucket.Update(0xc08410b300, 0xc0843a5870, 0x2f, 0x0, 0xc084
33b240, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/base/bucket.go:62 +0x57
github.com/couchbaselabs/sync_gateway/auth.(*Authenticator).getPrincipal(0xc08433b220, 0xc0843a5870, 0x2f, 0xab
fff0, 0x24, ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/auth/auth.go:128 +0xe8
github.com/couchbaselabs/sync_gateway/auth.(*Authenticator).GetUser(0xc08433b220, 0xc0843a5810, 0x24, 0x0, 0x0,
 ...)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/auth/auth.go:69 +0x85
github.com/couchbaselabs/sync_gateway/rest.(*handler).createUserSession(0xc08433cd80, 0xc08433b1c0, 0x2)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/session_api.go:126 +0x1e3
github.com/couchbaselabs/sync_gateway/rest.(*handler).invoke(0xc08433cd80, 0xac1e00, 0x0, 0x0)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/handler.go:159 +0x35f
github.com/couchbaselabs/sync_gateway/rest.func┬╖015(0x348638, 0xc084010d20, 0xc0843a6680)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/handler.go:86 +0x7f
net/http.HandlerFunc.ServeHTTP(0xc08415fbc0, 0x348638, 0xc084010d20, 0xc0843a6680)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1220 +0x43
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0840a1960, 0x348638, 0xc084010d20, 0xc0843a6680)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/vendor/src/github.
com/gorilla/mux/mux.go:86 +0x1df
github.com/couchbaselabs/sync_gateway/rest.func┬╖017(0x348638, 0xc084010d20, 0xc0843a6680)
        c:/jenkins/workspace/build_sync_gateway_103_win-2008-x64/app-under-test/sync_gateway/src/github.com/cou
chbaselabs/sync_gateway/rest/routing.go:203 +0xcb
net/http.HandlerFunc.ServeHTTP(0xc0840e9b80, 0x348638, 0xc084010d20, 0xc0843a6680)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1220 +0x43
net/http.serverHandler.ServeHTTP(0xc084195460, 0x348638, 0xc084010d20, 0xc0843a6680)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1597 +0x171
net/http.(*conn).serve(0xc08400c880)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1167 +0x7ba
created by net/http.(*Server).Serve
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/server.go:1644 +0x28e

goroutine 75 [select]:
net/http.(*persistConn).writeLoop(0xc08431a280)
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/transport.go:791 +0x274

created by net/http.(*Transport).dialConn
        C:/Users/ADMINI~1/AppData/Local/Temp/2/makerelease155988687/go/src/pkg/net/http/transport.go:529 +0x621
```

**Comment by adamcfraser** [comment]
Thanks @elevine  I've been able to reproduce in a unit test - planning to get a fix on master asap.

**Comment by tleyden** [comment]
@andreibaranouski can you add a functional test for this?

**Comment by tleyden** [comment]
@adamcfraser @snej so in this case, the emitted entry on the changes feed will look something like this?

```
{"results":[
{"seq":2,"id":"test2","changes":[{"rev":"1-e18422e6a82d0f2157d74b5dcf457997"}]}
],
{"seq": 0, "id":"", "err":"some error message", "changes":[]},
"last_seq":2}
```

are we 100% sure that this is compatible with couchbase lite clients?

**Comment by adamcfraser** [comment]
@tleyden No - the changes feed never emits the entry with error message.  The error is used as a trigger to terminate the continuous changes feed within the Sync Gateway.

**Comment by tleyden** [comment]
Merged in https://github.com/couchbase/sync_gateway/commit/9280842916ecfd8d8f8fd12690e7649a5437688a.

This should be fixed on the release/1.1.0 branch.
